### ISSUE TITLE
not sure I like the "my users" phrasing

### ISSUE THREAD

**Issue by shuckle16** [Widdershin/programmers-oath#58]
Really enjoy the oath, but I think saying "my users" sends the wrong message. "people who use the software to which I contribute" is not as pretty, but I think rephrasing is worth a discussion.

**Comment by mo-g** [comment]
Can you explain what you mean by 'wrong message'?

**Comment by shuckle16** [comment]
I think it gives the impression of ownership. to me it's fine to own the software, but not the users.

**Comment by Widdershin** [comment]
Hmmm, it's pretty common vernacular to refer to your product's users as "our users" in the circles I'm in, but I can see the awkward connotations of "my".

How about:

> I will respect the privacy of those who use my software, for their information is not disclosed to me that the world may know.

**Comment by mo-g** [comment]
Clean, the message is the same and it solves the issue.

**Comment by vassudanagunta** [comment]
Naw, keep "my  users". This is a humanistic oath. About connectedness. Think "my family", "my friends", "my world".  The important connection is between the programmer and the users ("my users" 👍), not between the programmer and the code ("my software" 👎).

**Comment by vassudanagunta** [comment]
@mo-g, you may consider this but a technical difference that doesn't really matter, but let me refer you to the Oath's own words:

> I will remember that programming is art as well as science, and that warmth, empathy and understanding may outweigh a clever algorithm or technical argument.

**Comment by pachunka** [comment]
I have to agree with @vassudanagunta - the language of each share appear to share the same technical meaning, but "my users" carries an additional emotional payload; one which makes the reader think twice about the between them and their users*.

* (tried to avoid bypassing the issue while crafting that sentence and I failed :j)

This doubt started because "My" appeared to imply ownership in the sense of control - e.g. My project, my servant - instead of the more compassionate relationship driven meaning of "My" - my neighbours, my community, my town.

Per tenet no.6 about writing code "for people", I think that little dose of discomfort on reading "my users" is a good thing.

The flash-of-thought that comes to mind when you think of even your most passive users as somehow being your responsibility - I feel like it goes away when "my users" becomes the more passive "the users of my software." Even "my software" has problems - the user thinks its _their_ software! :j

I have to apologise because this response is a bit rushed and I'm on my way out the door, but I felt this was important enough to add something to in case the issue is on its way to closing.

In short, I think the dissonance in the phrase "my users" is successfully avoided by the new phrasing, and I think that's a bad thing.

**Comment by EvanBurchard** [comment]
"users" throws the whole document into question honestly.

Tufte's phrasing on [this](https://devrant.com/rants/353877/there-are-only-two-industries-that-call-their-customers-users-illegal-drugs-and) isn't amazing, but it does highlight how casually devs can use exploitative language. Personally, I'd agree that it is that, and even if mildly so, the relationship has a history (and future) of exploitation that reinflates the term with that kind of  meaning.

**Comment by vassudanagunta** [comment]
@EvanBurchard  I've wrestled with the word "users" as well in a project of mine (a *Users First!* oath), but that is not the question of this issue. The question here is about the use of "my".  Perhaps open a new issue to discuss the world "users" vs "customers" vs ?.

There are a few problems with "customers". It is a very transactional word. It doesn't cover open source. And it today's world, much if not most of the time the users are not the customer, but the product. I've for many years had ideas about starting a movement to fight this (the *Users First* oath is a part of that), but I think that is beyond the scope of this Programmer's Oath. If this Oath included any statements against advertising, it would die a quick death as too many programmers have comfy incomes paid by advertising. 

*It is difficult to get a man to understand something when his salary depends upon his not understanding it.”*  
~Upton Sinclair

**Comment by pachunka** [comment]
@EvanBurchard The linked quote draws a cute analogy, but the term 'user' is well-understood to be shorthand for users of computers, users of software, etc.

I used to think it was a bit impersonal when the word started showing up everywhere, but what's done is done - and 'user account', 'username' etc. are some of the first things you see written on the screen when setting up a modern computer.

Your comment about exploitative language is a bit misplaced - it sounds like a jab intended for someone outside of this conversation that you hold frustration towards.

A collaborative project on github whose sole output is tailored language is hardly 'casual' use of language - and a highly-generic noun with a dubious history hardly amounts to being 'exploitative.'

And @vassudanagunta is right - it's an unusual scenario - the users of the product aren't necessarily traditional customers, or even consumers - can't consume something that's still there afterwards! =)

Yours non-casually,
A caffeine user.

**Comment by EvanBurchard** [comment]
@pachunka I'm weirded out by the sockpuppet/troll account. I guess I can block you? Never had to do that on github before, but man I don't have time for this kind of bullshit.

@vassudanagunta I just didn't think it was creating a new issue for the users thing since it seemed like already part of the discussion. Have a good one.

**Comment by pachunka** [comment]
@EvanBurchard This is my account; don't be too weirded out.

I didn't mean to troll. In truth, I didn't take the quote you linked out to very seriously, and it sounded to me like you were taking a jab at some unidentified "devs" who "casually use exploitative language", so I replied defensively instead of asking you for more details. I signed off with a joke-use of the word "user" to demonstrate a legitimate 3rd meaning of the word that the linked-to quote left out.

I'm interested to hear what you mean by "the relationship has a history (and future) of exploitation," if you're interested in going into it. My interpretation of your argument before was that the term "user" simply reminded you of its usage in the context of drugs - and that this brought an image to mind of a software-engineer as a drug-dealer - and that allowing such an image to be evoked in an oath amounted to casually-exploitative language. If the point you were making was different, I missed it, and I'm interested to hear about it.
